sklearn/utils/validation.py

```diff
@@ -374,6 +378,9 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
         if force_all_finite:
             _assert_all_finite(array)
 
+    if array.ndim == 1:
```

Comment:
The problem is that the current ``ensure_2d`` does the wrong thing. I think we should change the behavior of ``ensure_2d`` from reshaping to warning / erroring.